frontend/appflowy_flutter/lib/plugins/document/presentation/more/cubit/document_appearance_cubit.dart

```diff
@@ -1,28 +1,49 @@
 import 'package:appflowy/core/config/kv_keys.dart';
+import 'package:appflowy/util/color_to_hex_string.dart';
 import 'package:appflowy/workspace/application/settings/appearance/base_appearance.dart';
 import 'package:bloc/bloc.dart';
+import 'package:flutter/material.dart';
 import 'package:shared_preferences/shared_preferences.dart';
 
 class DocumentAppearance {
   const DocumentAppearance({
     required this.fontSize,
     required this.fontFamily,
+    this.cursorColor,
+    this.selectionColor,
     this.defaultTextDirection,
   });
 
   final double fontSize;
   final String fontFamily;
+  final Color? cursorColor;
+  final Color? selectionColor;
   final String? defaultTextDirection;
 
+  ///For nullable fields (like `cursorColor`),
+  /// use the corresponding `isNull` flag (like `cursorColorIsNull`) to explicitly set the field to `null`.
+  ///
+  /// This is necessary because simply passing `null` as the value does not distinguish between wanting to
+  /// set the field to `null` and not wanting to update the field at all.
```

Comment:
Couldn't we just assume `empty || null == null`, or am I missing something?

Reply:
The previous version of the code looks like

```dart
DocumentAppearance copyWith({
    double? fontSize,
    String? fontFamily,
    String? defaultTextDirection,
  }) {
    return DocumentAppearance(
      fontSize: fontSize ?? this.fontSize,
      fontFamily: fontFamily ?? this.fontFamily,
      defaultTextDirection: defaultTextDirection,
    );
  }
}
```

defaultTextDirection is a nullable field of `DocumentAppearance`. The problem is when we try to copywith, like

```dart
documentAppearance.copyWith(fontFamily: ***); 
```

What we want is to get a value with the new fontFamily along with all the previous field value. However, in this case, since we didn’t give any value to `defaultTextDirection`, so no matter what is its previous value, it will be set to null, which is not our intention.

Thus, I add these **isNull flags to distinguish the case we want to set a null value, with the case we don't want to change the previous value at all.